src/transaction/log_page_buffer.c

```diff
@@ -3361,6 +3364,250 @@ logpb_write_toflush_pages_to_archive (THREAD_ENTRY * thread_p)
+ *       │           │ So, EOF is appended at the prev_lsa.
+ *       │           ▼
+ *       │       _PARTIAL_FLUSHED_END_OF_LOG
+ *       │           │
+ *       │           │ It's done to append log pages containing an incomplete log record.
+ *       │           │ If there is no incomplete log record, just a page has been appended.
+ *       │           ▼
+ *       │       _PARTIAL_ENDED
+ *       │           │
+ *       │           │ Flush log pages to remove the EOF.
+ *       │           │ It appends the EOF at the append_lsa instead.
+ *       │           ▼
+ *       │       _PARTIAL_FLUSHED_ORIGINAL
+ *       │           │
+ *       │           │ Remove the EOF at the append_lsa on the memory buffer.
+ *       └───────────┘ This will be flushed in the next time.
+ *
+ * (*): It's done to append log pages containing an incomplete log record without flushing in the middle.
+ * 
+ *      See logpb_catchup_end_append() and logpb_flush_all_append_pages() as well for the complete picture.
+ *     
+ *   log_pgptr(in): a log page pointer to append
+ */
+static void
+logpb_catchup_start_append (THREAD_ENTRY * thread_p)
+{
+  assert (LOG_CS_OWN_WRITE_MODE (thread_p));
+
+  switch (log_Pb.partial_append.status)
+    {
+    case LOGPB_APPENDREC_SUCCESS:
+      log_Pb.partial_append.status = LOGPB_APPENDREC_IN_PROGRESS;
+      break;
+    case LOGPB_APPENDREC_IN_PROGRESS:
+    case LOGPB_APPENDREC_PARTIAL_FLUSHED_END_OF_LOG:
+      /* keep the status */
+      break;
+    default:
+      /* invalid state */
+      assert_release (false);
+    }
+}
+
+/*
+ * logpb_catchup_end_append - Finish appending a page during catch-up.
+ *      If the appended page has only parital record in it, keep partial appending status.
+ *      Otherwise, complete appending pages, in the partial appending case, or a page.
+ *        - Set the append_lsa and prev_lsa to valid ones.
+ *        - Remove the temporary EOF If log pages are flushed halfway through.
+ *        - Set the partial appending status to LOGPB_APPENDREC_SUCCESS.
+ * 
+ *    See logpb_catchup_start_append() for details of the status transition.
+ *
+ *   log_pgptr(in): a log page pointer appended
+ */
+static void
+logpb_catchup_end_append (THREAD_ENTRY * thread_p, const LOG_PAGE * const log_pgptr)
+{
+  assert (LOG_CS_OWN_WRITE_MODE (thread_p));
+
+  const bool have_only_partial_record_in_page = log_pgptr->hdr.offset == NULL_LOG_OFFSET;
+
+  if (have_only_partial_record_in_page)
+    {
+      // Do nothing. It will be finished up at a following call.
+      return;
+    }
+
+  assert (log_Gl.hdr.append_lsa.pageid == log_pgptr->hdr.logical_pageid);
+  log_Gl.hdr.append_lsa.offset = log_pgptr->hdr.offset;
+
+  LOG_RECORD_HEADER *log_rec = LOG_GET_LOG_RECORD_HEADER (log_pgptr, &log_Gl.hdr.append_lsa);
+  log_Gl.append.prev_lsa = log_rec->back_lsa;
+
+  switch (log_Pb.partial_append.status)
+    {
+    case LOGPB_APPENDREC_IN_PROGRESS:
+      /* success, fall through */
```

Comment:
Nit.
```suggestion
      /* success */
```
In my understanding of c/c++ "fall-through" is where a `case` label does not have a `break`.

https://en.cppreference.com/w/cpp/language/attributes/fallthrough

Reply:
Done. I agree. I just took the base of this code from `logpb_end_append()`.